### PR TITLE
Add Fulltext unicode tests

### LINKED ISSUE #31640: Ensure correct length calculation for strings in fulltext index
Let's consider emoji 4️⃣: it consists of 3 seperate codepoints. How should we treat it when building ngram index for example?

### PR DESCRIPTION
**Pull request by ghost**
### Changelog entry 

Add tests that cover fulltext search with ngram index applied to texts with unicode characters.

### Changelog category 

* Not for changelog (changelog entry is not required)

### Description for reviewers 

Added one unit test that directly tests how ngrams are built for unicode texts. And the other test checks whole flow: populating table, building index and querying with unicode characters in both data and queries.

### CODE REVIEW

File: ydb/core/base/ut/fulltext_ut.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "fulltext.h"
+#include "fulltext.cpp"
```
[the remark this answers is not in the document]
Comment (azevaykin) [reply ydb/core/base/ut/fulltext_ut.cpp]: +1

**Review by gridnevvvit — APPROVED** [review]
